Mathlib/Algebra/Order/Archimedean/Class.lean

```diff
@@ -452,3 +452,127 @@ theorem liftOrderHom_mk (f : M → α) (h : ∀ a b, mk a ≤ mk b → f a ≤ f
 end LiftHom
 
 end MulArchimedeanClass
+
+variable (M) in
+/-- Subtype of `MulArchimedeanClass` that removes the top element. -/
+@[to_additive ArchimedeanClass₀ "Subtype of `ArchimedeanClass` that removes the top element."]
+abbrev MulArchimedeanClass₁ := {A : MulArchimedeanClass M // A ≠ ⊤}
+
+namespace MulArchimedeanClass₁
+
+/-- Create a `MulArchimedeanClass₁` from a non-one element. -/
+@[to_additive "Create an `ArchimedeanClass₀` from a non-zero element."]
+def mk {a : M} (h : a ≠ 1) : MulArchimedeanClass₁ M :=
+  ⟨MulArchimedeanClass.mk a, MulArchimedeanClass.mk_eq_top_iff.not.mpr h⟩
+
+@[to_additive (attr := simp)]
+theorem val_mk {a : M} (h : a ≠ 1) : (mk h).val = MulArchimedeanClass.mk a := rfl
+
+@[to_additive]
+theorem mk_le_mk {a : M} (ha : a ≠ 1) {b : M} (hb : b ≠ 1) :
+    mk ha ≤ mk hb ↔ MulArchimedeanClass.mk a ≤ MulArchimedeanClass.mk b := .rfl
+
+@[to_additive]
+theorem mk_lt_mk {a : M} (ha : a ≠ 1) {b : M} (hb : b ≠ 1) :
+    mk ha < mk hb ↔ MulArchimedeanClass.mk a < MulArchimedeanClass.mk b := .rfl
+
+/-- An induction principle for `MulArchimedeanClass₁`. -/
+@[to_additive (attr := elab_as_elim) "An induction principle for `ArchimedeanClass₀`."]
+theorem ind {motive : MulArchimedeanClass₁ M → Prop}
+    (mk : ∀ a, (ha : a ≠ 1) → motive (.mk ha)) : ∀ x, motive x := by
+  intro ⟨A, hA⟩
+  induction A using MulArchimedeanClass.ind with | mk a
+  exact mk _ <| MulArchimedeanClass.mk_eq_top_iff.not.mp hA
```

Comment:
```suggestion
  simpa [MulArchimedeanClass₁]
```

Reply:
Oh, you're missing `MulArchimedean.forall`, the specialisation of `Quotient.forall`